src/app/stock-view/stock-view.component.ts

```diff
@@ -0,0 +1,93 @@
+import {ActivatedRoute} from "@angular/router";
+import {BaseChartDirective} from "ng2-charts";
+import {ChartConfiguration, ChartType} from "chart.js";
+
+@Component({
+  selector: 'app-stock-view',
+  standalone: true,
+  imports: [BaseChartDirective],
+  templateUrl: './stock-view.component.html',
+  styleUrl: './stock-view.component.css'
+})
+export class StockViewComponent {
+  @ViewChild(BaseChartDirective)
+  chart: BaseChartDirective | undefined;
+  stockHistory: ListingHistory[] = [];
+  stockId: number = 0;
+  stock: StockListing | null = null;
+  graphFilter: "d" | "m" | "y" | "5y" | "all" = "d";
+
+  chartData: ChartConfiguration["data"] = {
+    datasets: [
+      {
+        data: [],
+        label: 'Stock price history',
+        fill: "origin"
+      }
+    ],
+    labels: ["00:00", "04:00", "08:00", "12:00", "16:00", "20:00", "23:59"],
+  }
+  lineChartType: ChartType = 'line';
+  lineChartOptions: ChartConfiguration['options'] = {
+    responsive: true
+  }
+
+  constructor(private stockService: StockService, private route: ActivatedRoute) {
+    this.route.params.subscribe(params =>
+      this.stockId = params["id"]
+    )
+  }
+
+  async ngOnInit() {
+    this.stock = await this.stockService.getStockById(this.stockId);
+    await this.handleGraphFilter("d");
+  }
+
+  async handleGraphFilter(input: "d" | "m" | "y" | "5y" | "all") {
+    this.graphFilter = input;
+    let date = new Date();
+    let graphFilterValue: number | null = null;
+    switch(input) {
+      case "d":
+        date.setDate(date.getDate() - 1)
+        graphFilterValue = date.getTime();
+        this.chartData.labels = ["00:00", "04:00", "08:00", "12:00", "16:00", "20:00", "23:59"]
+        this.chart?.chart?.update()
+        break;
+      case "m":
+        date.setMonth(date.getMonth() - 1)
+        graphFilterValue = date.getTime();
+        this.chartData.labels = ["1.", "5.", "10.", "15.", "20.", "25.", "30."]
+        this.chart?.chart?.update()
+        break;
+      case "y":
+        date.setFullYear(date.getFullYear() - 1)
+        graphFilterValue = date.getTime();
+        this.chartData.labels = ['January', 'March', 'June', 'September', "December"]
+        break;
+      case "5y":
+        date.setFullYear(date.getFullYear() - 5)
+        graphFilterValue = date.getTime();
+        this.chartData.labels = [date.getFullYear(), date.getFullYear() + 1, date.getFullYear() + 2, date.getFullYear() + 3, date.getFullYear() + 4, date.getFullYear() + 5]
+        break;
+      case "all":
+        graphFilterValue = null
+        this.chartData.labels = [date.getFullYear() - 10, date.getFullYear() - 9, date.getFullYear()  - 8, date.getFullYear() - 7, date.getFullYear() - 6, date.getFullYear() - 5, date.getFullYear() - 4, date.getFullYear() - 3, date.getFullYear() - 2, date.getFullYear() - 1, date.getFullYear()]
+        break;
+    }
+
+    this.stockHistory = await this.stockService.getStockHistory(this.stockId, null, graphFilterValue);
```

Comment:
ovde sa cim cemo da zamenimo null, posto je to from, da li treba da bude vrednost danasnjeg dana - izbrani filter(d,m,g,5g....)

Reply:
U pravu si, "to" vrednost treba da bude null jer uvek treba da vraca do najskorije vrednosti. Popravljeno :D

Reply:
Super